### PR TITLE
ENH: Allow Iterable[Hashable] in drop_duplicates

### LINKED ISSUE #59237: ENH: Reduce type requirements for the subset parameter in drop_duplicates/duplicated
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I would like to pass a set to drop_duplicates like so:

```
df = get_some_df()
subset = {"column1", "column3"}
df_dropped = df.drop_duplicates(subset=subset)
```
According to type hints, `subset` is a `Hashable | Sequence[Hashable] | None`. The documentation says "column label or sequence of labels, optional". 

The problem is, I would like to pass a set of columns. The name `subset` suggests that should be ok. And it does work indeed. **However, a `set` is not a `Sequence`.**

Can the requirements be lowered? Maybe to `Collection`? (or even `Iterable`, but that might come with problems).

Looking at the code: https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/core/frame.py#L6935

It checks if the subset is `np.iterable`, then a `set` is created and `len(subset)` is called. This could be done with collections as well.

Am I missing anything or could this Sequence be make a Collection?

### Feature Description

If I am not mistaken, Sequence could simply be replaced with Collection in duplicated as well as drop_duplicates + in the docu.

### Alternative Solutions

no alt solution required

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kirill-bash**
- [x] closes #59237 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/frame.py
Comment (rhshadrach) [inline pandas/core/frame.py:23]: Why the alias?

Comment (rhshadrach) [reply pandas/core/frame.py:23]: Also, can you just use `set` instead of importing from `abc`?

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by rhshadrach** [comment]
Also - can you give this PR an informative title. Something like `ENH: Allow sets in drop_duplicates`.

File: pandas/core/frame.py
Comment (kirill-bash) [reply pandas/core/frame.py]: Good point! Updated.

File: pandas/core/frame.py
Comment (kirill-bash) [reply pandas/core/frame.py]: The alias was done by a hook when I tried to import Set from `abc` since it was ambiguous with `set`. I like your suggestion of just using `set` instead. Updated.

File: pandas/core/frame.py
Comment (Dr-Irv) [inline pandas/core/frame.py:6537]: I think you should just replace `Sequence[Hashable]` with `Iterable[Hashable]` here (and in the other overloads), because even a `dict` works now.

Also, I suggest making a separate PR in `pandas-stubs` once this PR is accepted.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
You should also update the docs to indicate that any iterable of labels is accepted.

File: pandas/core/frame.py
Comment (kirill-bash) [reply pandas/core/frame.py]: @Dr-Irv,

Updated. Could you specify which docs I should be updating to reflect these changes?

Also, what is the difference between `pandas` and `pandas-stubs`?
I am a new contributor and am happy to raise the PR there but would like to understand more about each repo.

**Review by Dr-Irv — APPROVED** [review]
I'm fine with this.  Will let someone else make the final call.

File: pandas/core/frame.py
Comment (Dr-Irv) [reply pandas/core/frame.py]: > Also, what is the difference between `pandas` and `pandas-stubs`? I am a new contributor and am happy to raise the PR there but would like to understand more about each repo.

`pandas` is the code that executes.

`pandas-stubs` is a set of typing declarations meant for users to use when type checking their code.  It is bundled within Visual Studio Code.   See https://github.com/pandas-dev/pandas-stubs/

File: pandas/core/frame.py
Comment (rhshadrach) [inline pandas/core/frame.py:6537]: No need for `set` anymore now that you've changed `Sequence` to `Iterable`, right?

File: doc/source/whatsnew/v3.0.0.rst
Comment (rhshadrach) [inline doc/source/whatsnew/v3.0.0.rst:54]: Same here - can remove `Set` now.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py
Comment (kirill-bash) [reply pandas/core/frame.py]: Thanks for clarifying. I will do as you suggested and open a PR there once this one is merged.

File: pandas/core/frame.py
Comment (kirill-bash) [reply pandas/core/frame.py]: That's true. Updated.

File: doc/source/whatsnew/v3.0.0.rst
Comment (kirill-bash) [reply doc/source/whatsnew/v3.0.0.rst]: Updated. Thanks!

**Review by rhshadrach — CHANGES_REQUESTED** [review]
CI is failing on code checks. I think it's the line `subset = self.columns # type: ignore[assignment]` where the `type: ignore` can now be removed.

**Review by rhshadrach — APPROVED** [review]
lgtm, can be merged on green.

**Comment by rhshadrach** [comment]
Thanks @kirill-bash!

**Comment by kirill-bash** [comment]
Appreciate your help, @rhshadrach!

**Pull request merged**
